File: frontend/src/types/connections.ts
export interface ConnectionSlot {
  id: string;
  project_id: string;
  name: string;
  owner: string;
  stable_port: number;
  target_environment_id: string;
  created_at: string;
  status: string;
}

export interface CreateConnectionPayload {
  name: string;
  owner: string;
  stable_port: number;
  target_environment_id: string;
  project_id: string;
}

export type UpdateConnectionPayload = Partial<CreateConnectionPayload>;

export interface ConnectionsResponse {
  connections: ConnectionSlot[];
}

File: backend/app/schemas/connections.py
from datetime import UTC, datetime

from pydantic import BaseModel, Field


class ConnectionSlotCreate(BaseModel):
    name: str = Field(min_length=1)
    owner: str = Field(min_length=1)
    stable_port: int = Field(ge=1, le=65535)
    target_environment_id: str = Field(min_length=1)
    project_id: str = "default"


class ConnectionSlotUpdate(BaseModel):
    name: str | None = Field(default=None, min_length=1)
    owner: str | None = Field(default=None, min_length=1)
    stable_port: int | None = Field(default=None, ge=1, le=65535)
    target_environment_id: str | None = Field(default=None, min_length=1)
    project_id: str | None = Field(default=None, min_length=1)


class ConnectionSlot(ConnectionSlotCreate):
    id: str
    created_at: str = Field(default_factory=lambda: datetime.now(UTC).isoformat())
    status: str = "active"


class ConnectionSlotListResponse(BaseModel):
    connections: list[ConnectionSlot]
